hudi-client/src/main/java/org/apache/hudi/io/storage/HoodieStorageWriterFactory.java

```diff
@@ -66,4 +67,21 @@
 
     return new HoodieParquetWriter<>(instantTime, path, parquetConfig, schema, sparkTaskContextSupplier);
   }
+
+  private static <T extends HoodieRecordPayload, R extends IndexedRecord> HoodieStorageWriter<R> newHFileStorageWriter(
+      String instantTime, Path path, HoodieWriteConfig config, Schema schema, HoodieTable hoodieTable,
+      SparkTaskContextSupplier sparkTaskContextSupplier) throws IOException {
+
+    BloomFilter filter = createBloomFilter(config);
```

Comment:
HFile allows adding BloomFilters using an interface. They are not always present (i.e. there is no AvroWriteSupport kind of implementation).

I think there are some benefits to using Hoodie's BloomFilter:
1. If we use HBase's implementation of BloomFilter, we will have to convert it to HoodieBloom filter as Hoodie code uses HoodieBloomFilter in functions
2. Advances in Hoodie's Bloom filter (compression, dynamic sizing) etc will not be available to HFile format
3. Common interface across all base file formats so easier to maintain in the long run and easier to update

Reply:
okay. I was thinking about dynamic sizing as well

Reply:
Nope. We are serializing the HoodieBloomFilter to bytes and saving this as a metadata block in HFile. HFile has support for adding custom named blocks of data which are loaded on demand (unlike Parquet Footer which is always read). 

HFile has its own interface BloomFilters which can also be implemented by HoodieBloomFilter. But that sounds overkill to me. 

As more base file formats are added, I think we should focus on common functionality so that we prevent code duplication.

Reply:
I think the dynamic bloom is reason enough. So good here